src/CSET/operators/mesoscale.py

```diff
@@ -0,0 +1,123 @@
+
+The diagnostics here are applicable at mesoscales and apply generally
+rather than for specific aspects of mesoscale meteorology (e.g. convection).
+For specific aspects, the user is referred to other modules available in
+CSET.
+
+"""
+
+import logging
+
+import iris
+import numpy as np
+from scipy.ndimage import gaussian_filter, uniform_filter
+
+from CSET.operators._utils import get_cube_yxcoordname
+
+
+def spatial_perturbation_field(
+    original_field: iris.cube.Cube,
+    Gaussian_filter: bool = True,
+    filter_scale: int = 40,
+) -> iris.cube.Cube:
+    """Calculate a spatial perturbation field.
+
+    Parameters
+    ----------
+    original_field: iris.cube.Cube
+        Iris cube containing data to smooth, supporting multiple dimensions
+        (at least two spatial dimensions must be supplied, i.e. 2D).
+    Gaussian_filter: boolean
+        Switch to determine if a Gaussian filter is applied.
+        If set to True a Gaussian filter is applied; if set to False
+        a Uniform filter is applied.
+        Default is True.
+    filter_scale: int
+        Scale at which to define the filter in grid boxes. If the
+        filter is a Gaussian convolution this value represents the
+        halfwidth of the Gaussian kernel.
+        Default is 40 grid points.
+
+    Returns
+    -------
+    An iris cube of the spatial perturbation field.
+
+    Return type
+    -----------
+    iris.cube.Cube
+
+    Notes
+    -----
+    In mesoscale meteorology the perturbation field is more important than the
+    balanced flows for process understanding. This function is designed
+    to create spatial perturbation fields based on smoothing with a Gaussian
+    kernel or a uniform kernel.
+
+    The kernels are defined by the filter_scale, which for mesoscale
+    perturbations should be between an approximate cloud separation distance,
+    of 30 km, and synoptic scale variations (1000 km). In practice any
+    value between these ranges should provided broadly consistent results (e.g.
+    [Flacketal2016]_). The Gaussian kernel will give greater importance to
+    areas closer to the event and will produce a smooth perturbation field.
+    The uniform kernel will produce a smooth perturbation field but will not
+    give local features as much prominence.
+
+    Caution should be applied to boundaries, particularly if the domain is of
+    variable resolution, as some numerical artifacts could be introduced.
+
+    References
+    ----------
+    .. [Flacketal2016] Flack, D.L.A., Plant, R.S., Gray, S.L., Lean, H.W.,
+       Keil, C. and Craig, G.C. (2016) "Characterisation of Convective
+       Regimes over the British Isles." Quarterly Journal of the Royal
+       Meteorological Society, vol. 142, 1541-1553. doi:10.1002/qj.2758
+
+    Examples
+    --------
+    >>> Temperature_perturbation = meso.spatial_perturbation_fields(Temp,
+                                   Gaussian_filter=True,filter_scale=40)
```

Comment:
```suggestion
                                   gaussian_filter=True,filter_scale=40)
```
Follow up from above change.